app/assets/javascripts/discourse/app/components/private-message-glyph.gjs

```diff
@@ -0,0 +1,20 @@
+import icon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+
+const PrivateMessageGlyph = <template>
+  {{#if @shouldShow}}
```

Comment:
we definitely don't have to change anything now but it's counter-intuitive to pass a `@shouldShow` arg instead of wrapping `<PrivateMessageGlyph />` in an `{{#if}}` 😃

Reply:
That's a question I frequently ask myself... Is visibility the responsibility of the wrapping component or the component itself?

Reply:
I'd say that depends whether the logic is (or makes sense to be) in the child component or not. In this case we're only passing in a flag from the outside.

```gjs
# bad
<SomeComponent if={{@visible}} />

# good
{{#if @visible}}<SomeComponent />{{/if}}

# also good
<SomeComponent @user={{this.user}} />
in ^ component:
{{#if (and @user.isStaff this.emergency)}}
  (actual contents)
{{/if}}

```

Reply:
Yes agreed this is more or less what I follow